tests/test_cache_memoize.py

```diff
@@ -350,3 +350,68 @@ def func_that_calls_runmeonce(*args):
         thread.join()
 
     assert len(calls_made) == 2
+
+
+class TestException(Exception):
+    pass
+
+
+class DerivedTestException(TestException):
+    pass
+
+
+class SecondTestException(Exception):
+    pass
+
+
+def test_dont_cache_exceptions():
```

Comment:
No other tests test that exceptions are propagated properly in default case. I think it's motivated since I am adding logic that can intercept exceptions.